portfolio/src/test/java/com/google/sps/DeleteCommentsTest.java

```diff
@@ -0,0 +1,151 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class DeleteCommentsTest {
+  private final HttpServletRequest postRequest = mock(HttpServletRequest.class);
+  private final HttpServletResponse postResponse = mock(HttpServletResponse.class);
+  private final DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+  private final DataServlet dataServlet = new DataServlet();
+  private final DeleteComments deleteComments = new DeleteComments();
+  private static final String COMMENT_ONE = "Test Comment #1";
+  private static final String COMMENT_TWO = "Test Comment #2";
+  private static final long TIMESTAMP_ONE = 0;
+  private static final long TIMESTAMP_TWO = 1;
+  private static final String NUM_COMMENTS_TO_DELETE = "delete-num";
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+  
+  private int getNumberOfEntriesInDatastore() {
+    return ds.prepare(new Query(dataServlet.COMMENT_TABLE_NAME)).countEntities(withLimit(10));
+  }
+
+  private void addEntityToDatastore(String commentText, long timestamp) {
+    Entity comment = new Entity(dataServlet.COMMENT_TABLE_NAME);
+    comment.setProperty(dataServlet.COMMENT_COLUMN_NAME, commentText);
+    comment.setProperty(dataServlet.TIMESTAMP_COLUMN_NAME, timestamp);
+    ds.put(comment);
+  }
+
+  @Test
+  public void testDeleteSomeComments() {
+    int numCommentsToDelete = 1;
+    addEntityToDatastore(COMMENT_ONE, TIMESTAMP_ONE);
+    addEntityToDatastore(COMMENT_TWO, TIMESTAMP_TWO);
+
+    when(postRequest.getParameter(NUM_COMMENTS_TO_DELETE))
+        .thenReturn(Integer.toString(numCommentsToDelete));
+        
+    try {
+        deleteComments.doPost(postRequest, postResponse);
+    } catch(Exception e) {
+        System.out.println("Error caught!");
+        System.out.println(e);
+    }
+
+    assertThat(getNumberOfEntriesInDatastore()).isEqualTo(1);
+  }
+
+  @Test
+  public void testDeleteAllComments() {
```

Comment:
Alternatively instead of modifying the try-catch block, you can instead do this:
```suggestion
  public void testDeleteAllComments() throws IOException {
```
Now you wouldn't need to worry about catching the exception, and I suspect when you're running tests, if this test throws the Exception, the testing framework will handle this better.

Reply:
thanks for this comment! it was really helpful!!